### ISSUE TITLE
Dead loop, caused by white spaces in style. Please fix.

### ISSUE THREAD

**Issue by WilliamXi** [robolectric/robolectric#3198]
### Description

I got a dead loop when running a robolectric test. Debug into the code I found below method should return true but it returns false due to white spaces in style item tag.
[AttributeResource.isResourceReferences(String value)](https://github.com/robolectric/robolectric/blob/master/resources/src/main/java/org/robolectric/res/AttributeResource.java) is called with value = `"\n                      @style/another_style       "`

the dead loop happens in [org.robolectric.shadow.ShadowAssetManager.java,](https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowAssetManager.java) in the while statement (line 713), there isn't match to any conditions.

I think the fix should be in somewhere which stripping out white spaces,  instead of in method isResourceReferences(String) and isStyleReference(String). 

### Steps to Reproduce

The style has below format: 
```
<style name="AppTheme" parent="@android:style/Theme.Holo">
   <item name="sample_style">
            @style/another_style
   </item>
...
```
the layout file is:
```
 <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:id="@+id/sample_id"
    style="?attr/sample_style">
```   
attr.xml is:
```
 <attr name="sample_style" format="reference" />
``` 
AndroidManifest.xml is:
```
<application
       android:allowBackup="true"
      android:icon="@drawable/ic_launcher"
      android:theme="@style/AppTheme" >
``` 
In unit test, I'm testing CustomArrayAdapter

```    @Test
    public void usesTextViewResourceIdToSetTextWithinListItemView() throws Exception {
        Activity activity = Robolectric.setupActivity(MyActivity.class);

        // NOTE: instead of RuntimeEnvironment.application, activity instance is passed to ListView, 
       // so that style reference can be found.
        ListView parent = new ListView(activity);  
        CustomArrayAdapter<String> arrayAdapter 
                = new CustomArrayAdapter<>(activity, R.layout.main, R.id.title, new String[] { "first value" });
        View listItemView = arrayAdapter.getView(0, null, parent);
        TextView titleTextView = (TextView) listItemView.findViewById(R.id.title);
        assertEquals("first value", titleTextView.getText().toString());
    }
```
### Robolectric & Android Version
Robolectric: 3.3.2
Android: 23

**Comment by WilliamXi** [comment]
Just an update - After using single line format, the issue is worked around. But robolectric code should tolerate white spaces, because 're-format code' command will wrap long items into 3 lines format.
```
<item name="sample_style">@style/another_style</item>
```

**Comment by WilliamXi** [comment]
Hi, is this issue actually checked by any Robolectric developer? Please fix it. Please!

**Comment by jongerrish** [comment]
Triaged to 3.5 - would you be up for sending a pull request to fix this?

**Comment by WilliamXi** [comment]
Hi @xian and @nll , I tested 3.5.1, #3475 didn't fix the dead loop. 
The reason is "\n    " in below.  white spaces should be trimmed for <item>'s value.

 <item name="sample_style">
            @style/another_style
   </item>

**Comment by jongerrish** [comment]
Should be fixed if you use Robolectric 4.0 with binary resource mode enabled. Please give it a try and reopen if its still an issue there.
